lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1622,6 +1624,93 @@ def get_configured_disk_size(self, expected_disk_spec):
+        4. vmdk_folder: The "/path/to/" portion of the string (os.path.dirname equivalent)
+        """
+        try:
+            datastore_name = re.match(r'^\[(.*?)\]', vmdk_path, re.DOTALL).groups()[0]
+            vmdk_fullpath = re.match(r'\[.*?\] (.*)$', vmdk_path).groups()[0]
+            vmdk_filename = os.path.basename(vmdk_fullpath)
+            vmdk_folder = os.path.dirname(vmdk_fullpath)
+        except (IndexError, AttributeError) as e:
+            raise RuntimeError("Bad path for filename disk vmdk image")
+
+        return (datastore_name, vmdk_fullpath, vmdk_filename, vmdk_folder)
+
+    def find_vmdk(self, vmdk_path):
+        """
+        Takes a vsphere datastore path in the format
+
+            [datastore_name] /path/to/file.vmdk
+
+        Returns vsphere file object or raises RuntimeError
+        """
+        datastore_name, vmdk_fullpath, vmdk_filename, vmdk_folder \
+            = self.vmdk_disk_path_split(vmdk_path)
+
+        datastore = self.cache.find_obj(self.content,
+                                        [vim.Datastore],
+                                        datastore_name)
+        browser = datastore.browser
+        detail_query = vim.host.DatastoreBrowser.FileInfo.Details(
+            fileOwner=True,
+            fileSize=True,
+            fileType=True,
+            modification=True
+        )
+        searchSpec = vim.host.DatastoreBrowser.SearchSpec(
+            details=detail_query,
+            matchPattern=[vmdk_filename],
+            searchCaseInsensitive=True,
+        )
+        search_res = browser.SearchSubFolders(
+            datastorePath="[" + datastore_name + "]",
+            searchSpec=searchSpec
+        )
+        while search_res.info.state not in [vim.TaskInfo.State.success,
+                                            vim.TaskInfo.State.error]:
+            time.sleep(1)
+        if search_res.info.result is None:
+            raise RuntimeError("No valid disk vmdk image found")
+        target_folder_path = "[" + datastore_name + "]" + " " + vmdk_folder + '/'
+        for result in search_res.info.result:
+            for f in getattr(result, 'file'):
+                if f.path == vmdk_filename and \
+                   result.folderPath == target_folder_path:
+                    return f
+        raise RuntimeError("No vmdk file found")
+
+    def add_existing_vmdk(self, vm_obj, expected_disk_spec, diskspec, scsi_ctl):
+        """
+        Adds vmdk file described by expected_disk_spec['filename'], retrieves the file
+        information and adds the correct spec to self.configspec.deviceChange.
+        """
+        filename = expected_disk_spec['filename']
+        # if this is a new disk, or the disk filenames are different
+        if (vm_obj and diskspec.device.backing.fileName != filename) or (vm_obj is None):
+            vmdk_file = self.find_vmdk(expected_disk_spec['filename'])
+            datastore_name, vmdk_fullpath, vmdk_filename, vmdk_folder \
```

Comment:
The results of vmdk_disk_path_split are not used anywhere here (they are used inside find_vmdk, but it calls the routine directly.